compress.cc

```diff
@@ -154,6 +156,51 @@ auto zstd_processor::get_algorithm() const -> algorithm {
     return algorithm::zstd;
 }
 
+const std::string_view DICTIONARY_OPTION = ".dictionary.";
+
+[[maybe_unused]]
+static std::map<sstring, sstring> dict_as_options(std::span<const std::byte> d) {
+    std::map<sstring, sstring> result;
+    const size_t max_part_size = std::numeric_limits<uint16_t>::max() - 1;
+    while (!d.empty()) {
+        auto this_part_size = std::min(max_part_size, d.size());
+        auto part_name = fmt::format("{}{:08}", DICTIONARY_OPTION, result.size());
+        auto part = d.subspan(0, this_part_size);
+        auto part_as_string = std::string_view(reinterpret_cast<const char*>(part.data()), part.size());
+        result.emplace(part_name, part_as_string);
+        d = d.subspan(this_part_size);
+    }
+    return result;
+}
+
+[[maybe_unused]]
+static std::optional<std::vector<std::byte>> dict_from_options(const sstables::compression& c) {
+    std::map<int, bytes_view> parts;
+    for (const auto& [k, v] : c.options.elements) {
+        auto k_str = sstring(k.value.begin(), k.value.end());
+        if (k_str.starts_with(DICTIONARY_OPTION)) {
```

Comment:
Please document the format additions in docs/dev.

Reply:
Done.